### ISSUE TITLE
Rectangle rotation doesn't render

### ISSUE THREAD

**Issue by miatang13** [penrose/penrose#478]
**Describe the bug**
When we pass in arguments for `rotation` for the shape `rectangle`, it does not get rotated on the screen. 

**To Reproduce**
```
forall Set x {
    x.icon = Rectangle {
        rotation: // tried with values (0, 60)
    }
}
```

**Expected behavior**
Have the rectangle be rotated.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - Macbook Pro

**Smartphone (please complete the following information):**
NA

**Additional context**
NA

**Comment by maxkrieger** [comment]
`rotation` is not defined in most shapeDefs:

https://github.com/penrose/penrose/blob/main/packages/core/src/renderer/ShapeDef.ts#L163-L176

It is only present in `square`, `image`, `arrow`, but yeah, they won't render anyway. We should decide whether rotation is a universal property and update shapeDefs accordingly.

**Comment by joshpoll** [comment]
Hmm this is definitely problematic as e.g. `Image` is considered a rect-like for constraints, but it can be rotated. Many of the constraints currently assume rect-likes are axis-aligned.

**Comment by cmumatt** [comment]
We fixed this but had performance problems in CI afterward that were slowing down development.  The thinking at the time is the test suite uses a lot of rectangles; consequently, the slowdown was magnified.  

Due to other important items in play at the time (including performance improvements), we temporarily removed the fix via PR# #803.  At a later date we should re-integrate the fix and address any performance issues that continue to exist.

**Comment by wodeni** [comment]
Closing as I can't repro this issue now.
